settings.go

```diff
@@ -1,16 +1,22 @@
 package main
 
 import (
+	"encoding/json"
 	"fmt"
 	"regexp"
 	"strings"
 
 	mapset "github.com/deckarep/golang-set/v2"
 	"github.com/kubewarden/gjson"
 	kubewarden "github.com/kubewarden/policy-sdk-go"
-	easyjson "github.com/mailru/easyjson"
 )
 
+type RawSettings struct {
+	DeniedLabels      []string          `json:"denied_labels"`
+	MandatoryLabels   []string          `json:"mandatory_labels"`
+	ConstrainedLabels map[string]string `json:"constrained_labels"`
+}
```

Comment:
We should merge `RawSettings` with `Settings`, this was just a workaround caused by easyjson not supporting properly `mapset`. Can you give a shot at unmarshalling the json straight into `Settings`?

Reply:
refactored in: [c322882](https://github.com/kubewarden/safe-labels-policy/commit/c322882e2d9be13787ade663d8bc50c50980055c).
Implemented `json.Unmarshaler` for `Settings`, but kept an anonymous struct similar to `RawSettings`to simplify the unmarshalling.